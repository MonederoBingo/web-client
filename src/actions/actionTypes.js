export const BEGIN_AJAX_CALL = 'BEGIN_AJAX_CALL';
export const BEGIN_AJAX_ERROR = 'BEGIN_AJAX_ERROR';

export const LOAD_AUTHORS_SUCCESS = 'LOAD_AUTHORS_SUCCESS';

export const LOAD_JOGGING_TIMES_SUCCESS = 'LOAD_JOGGING_TIMES_SUCCESS';
export const CREATE_JOGGING_TIME_SUCCESS = 'CREATE_JOGGING_TIME_SUCCESS';
export const UPDATE_JOGGING_TIME_SUCCESS = 'UPDATE_JOGGING_TIME_SUCCESS';
export const DELETE_JOGGING_TIME_SUCCESS = 'DELETE_JOGGING_TIME_SUCCESS';
export const CLEAR_JOGGING_TIMES_SUCCESS = 'CLEAR_JOGGING_TIMES_SUCCESS';

export const GET_JOGGING_TIMES_BY_USER_SUCCESS = 'GET_JOGGING_TIMES_BY_USER_SUCCESS';
export const GET_JOGGING_TIMES_BY_USER_ERROR = 'GET_JOGGING_TIMES_BY_USER_ERROR';
export const CREATE_JOGGING_TIME_BY_USER_SUCCESS = 'CREATE_JOGGING_TIME_BY_USER_SUCCESS';
export const UPDATE_JOGGING_TIME_BY_USER_SUCCESS = 'UPDATE_JOGGING_TIME_BY_USER_SUCCESS';
export const DELETE_JOGGING_TIME_BY_USER_SUCCESS = 'DELETE_JOGGING_TIME_BY_USER_SUCCESS';

export const REGISTER_USER_SUCCESS = 'REGISTER_USER_SUCCESS';
export const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
export const GET_USER_ROLE_SUCCESS = 'GET_USER_ROLE_SUCCESS';
export const LOAD_USERS_SUCCESS = 'LOAD_USERS_SUCCESS';
export const LOAD_USERS_ERROR = 'LOAD_USERS_ERROR';

export const CHANGE_PASSWORD_SUCCESS = 'CHANGE_PASSWORD_SUCCESS';
export const CHANGE_PASSWORD_BY_USER_SUCCESS = 'CHANGE_PASSWORD_BY_USER_SUCCESS';

export const AUTHENTICATED_USER_SUCCESS = 'AUTHENTICATED_USER_SUCCESS';

export const CREATE_USER_SUCCESS = 'CREATE_USER_SUCCESS';
export const UPDATE_USER_SUCCESS = 'UPDATE_USER_SUCCESS';
export const DELETE_USER_SUCCESS = 'DELETE_USER_SUCCESS';

export const LOAD_AVG_SPEED_REPORT_SUCCESS = 'LOAD_AVG_SPEED_REPORT_SUCCESS';
