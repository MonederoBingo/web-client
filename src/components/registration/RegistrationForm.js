import React from 'react';
import TextInput from '../common/TextInput';
import PasswordInput from '../common/PasswordInput';

const RegistrationForm = ({user, onSave, onChange, onCancel, saving, errors}) => {
  let actionsStyle = {
    marginRight: 20,
    marginTop: 10
  };

  let submitStyle = {
    marginTop: 10,
    marginRight: 50
  };

  return (
    <form>
      <TextInput
        name="companyName"
        label="Company Name"
        placeholder="Company name"
        value=""
        onChange={() => {}}
        error=""/>

      <TextInput
        name="email"
        label="Email"
        placeholder="Email"
        value={user.email}
        onChange={onChange}
        error={errors.email}/>

      <PasswordInput
        name="password"
        label="Password"
        placeholder="Password"
        value={user.password}
        onChange={onChange}
        error={errors.password}/>

      <input
        type="submit"
        disabled={saving}
        value={saving ? 'Saving...' : 'Save'}
        className="btn btn-primary"
        onClick={onSave}
        style={submitStyle}/>

      <input
        id="cancel"
        type="submit"
        value="Cancel"
        className="btn btn-default"
        onClick={onCancel}
        style={actionsStyle}/>
    </form>
  );
};

export default RegistrationForm;
