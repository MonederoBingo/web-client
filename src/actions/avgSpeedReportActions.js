/*eslint-disable import/no-duplicates*/
import * as types from './actionTypes';
import joggingTimeApi from '../api/joggingTimeApi';
import {beginAjaxCall} from './ajaxStatusActions';
import {ajaxCallError} from './ajaxStatusActions';

export function loadAvgSpeedReportSuccess(avgSpeedPerWeek) {
  return {type: types.LOAD_AVG_SPEED_REPORT_SUCCESS, avgSpeedPerWeek};
}

export function loadAvgSpeedReport() {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return joggingTimeApi.loadAvgSpeedReport().then(serviceResult => {
      dispatch(loadAvgSpeedReportSuccess(JSON.parse(serviceResult.object)));
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}
