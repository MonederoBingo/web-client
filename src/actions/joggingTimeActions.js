/*eslint-disable import/no-duplicates*/
import * as types from './actionTypes';
import joggingTimeApi from '../api/joggingTimeApi';
import {beginAjaxCall} from './ajaxStatusActions';
import {ajaxCallError} from './ajaxStatusActions';

export function loadJoggingTimeSuccess(joggingTimes) {
  return {type: types.LOAD_JOGGING_TIMES_SUCCESS, joggingTimes};
}
export function createJoggingTimeSuccess(joggingTime) {
  return {type: types.CREATE_JOGGING_TIME_SUCCESS, joggingTime};
}
export function updateJoggingTimeSuccess(joggingTime) {
  return {type: types.UPDATE_JOGGING_TIME_SUCCESS, joggingTime};
}
export function deleteJoggingTimeSuccess(joggingTime) {
  return {type: types.DELETE_JOGGING_TIME_SUCCESS, joggingTime};
}

export function loadJoggingTimes(filter) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return joggingTimeApi.getAllJoggingTimes(filter).then(serviceResult => {
      dispatch(loadJoggingTimeSuccess(JSON.parse(serviceResult.object)));
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}

export function createJoggingTime(joggingTime) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return new Promise((resolve, reject) => {
      return joggingTimeApi.createJoggingTime(joggingTime).then(serviceResult => {
        if (serviceResult.success) {
          joggingTime.jogging_times_id = serviceResult.object;
          dispatch(createJoggingTimeSuccess(joggingTime));
          resolve();
        } else {
          dispatch(ajaxCallError());
          reject(serviceResult.message);
        }
      }).catch(error => {
        dispatch(ajaxCallError());
        reject(error);
      });
    });
  };
}

export function createJoggingTimeForUser(joggingTime, user) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return new Promise((resolve, reject) => {
      return joggingTimeApi.createJoggingTimeForUser(joggingTime, user).then(serviceResult => {
        if (serviceResult.success) {
          joggingTime.jogging_times_id = serviceResult.object;
          dispatch(createJoggingTimeSuccess(joggingTime));
          resolve();
        } else {
          dispatch(ajaxCallError());
          reject(serviceResult.message);
        }
      }).catch(error => {
        dispatch(ajaxCallError());
        reject(error);
      });
    });
  };
}

export function updateJoggingTime(joggingTime) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return new Promise((resolve, reject) => {
      return joggingTimeApi.updateJoggingTime(joggingTime).then(serviceResult => {
        if (serviceResult.success) {
          dispatch(updateJoggingTimeSuccess(joggingTime));
          resolve();
        } else {
          dispatch(ajaxCallError());
          reject(serviceResult.message);
        }
      }).catch(error => {
        dispatch(ajaxCallError());
        reject(error);
      });
    });
  };
}

export function deleteJoggingTime(joggingTime) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return joggingTimeApi.deleteJoggingTime(joggingTime).then(() => {
        dispatch(deleteJoggingTimeSuccess(joggingTime));
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}
