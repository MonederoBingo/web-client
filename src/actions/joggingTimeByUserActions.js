/*eslint-disable import/no-duplicates*/
import * as types from './actionTypes';
import joggingTimeApi from '../api/joggingTimeApi';
import {beginAjaxCall} from './ajaxStatusActions';
import {ajaxCallError} from './ajaxStatusActions';

export function getJoggingTimeByUserSuccess(joggingTimesByUser) {
  return {type: types.GET_JOGGING_TIMES_BY_USER_SUCCESS, joggingTimesByUser};
}
export function getJoggingTimeByUserError() {
  return {type: types.GET_JOGGING_TIMES_BY_USER_ERROR};
}
export function createJoggingTimeByUserSuccess(joggingTimesByUser) {
  return {type: types.CREATE_JOGGING_TIME_BY_USER_SUCCESS, joggingTimesByUser};
}
export function updateJoggingTimeByUserSuccess(joggingTimesByUser) {
  return {type: types.UPDATE_JOGGING_TIME_BY_USER_SUCCESS, joggingTimesByUser};
}
export function deleteJoggingTimeByUserSuccess(joggingTimesByUser) {
  return {type: types.DELETE_JOGGING_TIME_BY_USER_SUCCESS, joggingTimesByUser};
}

export function getJoggingTimesByUser(user, dateRangeFilter) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return joggingTimeApi.getJoggingTimesByUser(user, dateRangeFilter).then(serviceResult => {
      dispatch(getJoggingTimeByUserSuccess(JSON.parse(serviceResult.object)));
    }).catch(error => {
      dispatch(ajaxCallError());
      dispatch(getJoggingTimeByUserError());
      throw(error);
    });
  };
}

export function createJoggingTimeForUser(joggingTime, user) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return new Promise((resolve, reject) => {
      return joggingTimeApi.createJoggingTimeForUser(joggingTime, user).then(serviceResult => {
        if (serviceResult.success) {
          joggingTime.jogging_times_id = serviceResult.object;
          dispatch(createJoggingTimeByUserSuccess(joggingTime));
          resolve();
        } else {
          dispatch(ajaxCallError());
          reject(serviceResult.message);
        }
      }).catch(error => {
        dispatch(ajaxCallError());
        reject(error);
      });
    });
  };
}

export function updateJoggingTimeForUser(joggingTime, user) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return new Promise((resolve, reject) => {
      return joggingTimeApi.updateJoggingTimeForUser(joggingTime, user).then(serviceResult => {
        if (serviceResult.success) {
          dispatch(updateJoggingTimeByUserSuccess(joggingTime));
          resolve();
        } else {
          dispatch(ajaxCallError());
          reject(serviceResult.message);
        }
      }).catch(error => {
        dispatch(ajaxCallError());
        reject(error);
      });
    });
  };
}

export function deleteJoggingTimeForUser(joggingTime, user) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return joggingTimeApi.deleteJoggingTimeForUser(joggingTime, user).then(() => {
      dispatch(deleteJoggingTimeByUserSuccess(joggingTime));
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}
