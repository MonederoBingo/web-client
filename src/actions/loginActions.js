/*eslint-disable import/no-duplicates*/
import * as types from './actionTypes';
import * as authorizationApi from '../api/authorizationApi';
import {beginAjaxCall} from './ajaxStatusActions';
import {ajaxCallError} from './ajaxStatusActions';

export function loginUserSuccess(serviceResult) {
  return {type: types.LOGIN_USER_SUCCESS, serviceResult};
}

export function authenticatedUserSuccess(authInfo) {
  return {type: types.AUTHENTICATED_USER_SUCCESS, authInfo};
}

export function getUserRoleSuccess(serviceResult) {
  return {type: types.GET_USER_ROLE_SUCCESS, serviceResult};
}

export function clearJoggingTimesSuccess() {
  return {type: types.CLEAR_JOGGING_TIMES_SUCCESS};
}

let loginResult = {
  success: true,
  message: '',
  object: {}
};

function refresh(dispatch, token) {
  return authorizationApi.refreshToken(token).then(response => {
    sessionStorage.setItem('access_token', response.access_token);
    sessionStorage.setItem('expires_in', response.expires_in);
    sessionStorage.setItem('refresh_token', response.refresh_token);
    dispatch(authenticatedUserSuccess({authenticated: true, expires_in: parseInt(response.expires_in)}));
    dispatch(loginUserSuccess(loginResult));
  }).catch(error => {
    dispatch(ajaxCallError());
    throw(error);
  });
}

export function loginUser(user) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return authorizationApi.getAccessToken(user).then(response => {
      return refresh(dispatch, response.refresh_token);
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}

export function refreshToken() {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return refresh(dispatch, sessionStorage.getItem('refresh_token'));
  };
}

let userRoleResult = {
  success: true,
  message: '',
  object: {}
};

export function getUserRole(user) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return authorizationApi.getUserRole().then(response => {
      sessionStorage.setItem('role', response.authorities[0].authority);
      sessionStorage.setItem('name', response.name);
      dispatch(getUserRoleSuccess(userRoleResult));
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}

export function clearJoggingTimes() {
  return function (dispatch) {
    dispatch(clearJoggingTimesSuccess());
  };
}
