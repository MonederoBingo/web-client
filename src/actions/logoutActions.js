/*eslint-disable import/no-duplicates*/
import * as types from "./actionTypes";

export function authenticatedUserSuccess(authInfo) {
  return {type: types.AUTHENTICATED_USER_SUCCESS, authInfo};
}

let authInfo = {
  authenticated: false,
  expires_in: 0
};
export function logoutUser() {
  return function (dispatch) {
    sessionStorage.removeItem("access_token");
    sessionStorage.removeItem("role");
    sessionStorage.removeItem("name");
    sessionStorage.removeItem("expires_in");
    sessionStorage.removeItem("refresh_token");
    sessionStorage.removeItem("expires_in");
    dispatch(authenticatedUserSuccess(authInfo));
  };
}
