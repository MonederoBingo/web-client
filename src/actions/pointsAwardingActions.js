import * as types from './actionTypes';
import {beginAjaxCall} from './ajaxStatusActions';
import {ajaxCallError} from './ajaxStatusActions';
import * as pointsAwardingApi from '../api/pointsAwardingApi';

export function awardPointsSuccess(pointsAwarding) {
  return {type: types.AWARD_POINTS_SUCCESS, pointsAwarding};
}

export function awardPoints(pointsAwarding) {
  return function (dispatch) {
    dispatch(beginAjaxCall());
    return new Promise((resolve, reject) => {
      pointsAwardingApi.awardPoints(pointsAwarding).then(serviceResult => {
        if(serviceResult.success) {
          dispatch({type: 'POINTS_AWARDING_SUCCESS', serviceResult});

        } else {
          dispatch(ajaxCallError());
        }
      });
      resolve();
    });
  };
}
