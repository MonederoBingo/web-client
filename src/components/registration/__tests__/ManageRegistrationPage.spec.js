import React from 'react';
import expect from 'jest-matchers';
import ConnectedManageRegistrationPage, { ManageRegistrationPage } from '../ManageRegistrationPage';
import RegistrationForm from '../RegistrationForm';
import { shallow, mount } from 'enzyme';
import * as pointsAwardingActions from "../../../actions/pointsAwardingActions";
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

describe("ManageRegistrationPage", () => {
  let component;
  const mockStore = configureStore();
  beforeEach(() => {
    component = shallow(<ManageRegistrationPage/>);
  });
  function withEvent(name, value) {
    const event = {
      target: {
        name: name,
        value: value,
      }
    };
    return event;
  }
  it("should render initial layout", () => {
    // then
    expect(component.find('div').children().nodes).toMatchSnapshot();
  });
  it("should call updateFieldsState method in form", () => {
    // given
    jest.spyOn(ManageRegistrationPage.prototype, 'updateUserState');
    const form = shallow(<ManageRegistrationPage/>).find('RegistrationForm');

    // when
    form.props().onChange(withEvent("", ""));

    // then
    expect(ManageRegistrationPage.prototype.updateUserState).toHaveBeenCalled();
  });
  it("should pass 'saving' from state to form", () => {
    // when
    component.setState({saving: true});

    // then
    expect(component.find('RegistrationForm').props().saving).toBe(true);
  });
  it("should call saveUser method in form's onSave", () => {
    // given
    const spy = jest.spyOn(ManageRegistrationPage.prototype, 'saveUser');
    const actions = {
      createUser: () => {
        return new Promise((resolve) => {
          resolve();
        });
      }
    };
    const comp = shallow(<ManageRegistrationPage actions={actions}/>);
    const form = comp.find('RegistrationForm');
    const user = {
      email: 'email',
      password: 'password'
    };
    comp.setState({user});

    // when
    form.props().onSave({preventDefault: () => {} });

    // then
    expect(spy).toHaveBeenCalled();
  });
  // it("should pass pointsAwarding object from state to form", () => {
  //   // when
  //   component.setState({pointsAwarding: {}});
  //
  //   // then
  //   expect(component.find('RegistrationForm').props().pointsAwarding).toEqual({});
  // });
  // it("should update phone number in state", () => {
  //   // given
  //   const form = component.find('RegistrationForm');
  //   const event = withEvent("phoneNumber", "345");
  //
  //   // when
  //   form.props().onChange(event);
  //
  //   // then
  //   expect(component.instance().state.pointsAwarding.phoneNumber).toEqual("345");
  // });
  // it("should update amount in state", () => {
  //   // given
  //   const form = component.find('RegistrationForm');
  //   const event = withEvent("amount", 100);
  //
  //   // when
  //   form.props().onChange(event);
  //
  //   // then
  //   expect(component.instance().state.pointsAwarding.amount).toEqual(100);
  // });
  // it("should update sale key in state", () => {
  //   // given
  //   const form = component.find('RegistrationForm');
  //   const event = withEvent("saleKey", "key");
  //
  //   // when
  //   form.props().onChange(event);
  //
  //   // then
  //   expect(component.instance().state.pointsAwarding.saleKey).toEqual("key");
  // });
  // it("should map award points actions to props", () => {
  //   // given
  //   const connectedComponent = mount( <Provider store={mockStore()}><ConnectedManageRegistrationPage /></Provider> );
  //
  //   // when
  //   const component = connectedComponent.find('ManageRegistrationPage');
  //
  //   // then
  //   expect(component.props().actions.awardPoints).toBeInstanceOf(Function);
  // });
  // it("should call award points action on submit", () => {
  //   // given
  //   const connectedComponent = mount( <Provider store={mockStore()}><ConnectedManageRegistrationPage /></Provider> );
  //   const component = connectedComponent.find('ManageRegistrationPage');
  //   const form = component.find('RegistrationForm');
  //   component.props().actions.awardPoints = jest.fn();
  //
  //   // when
  //   form.props().onSubmit();
  //
  //   // then
  //   expect(component.props().actions.awardPoints).toHaveBeenCalled();
  // });
});
