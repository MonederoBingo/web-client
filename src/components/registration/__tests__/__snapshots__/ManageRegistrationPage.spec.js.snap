// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageRegistrationPage should render initial layout 1`] = `
Array [
  <h3>
    <strong>
      Registration
    </strong>
  </h3>,
  <RegistrationForm
    errors={Object {}}
    onCancel={[Function]}
    onChange={[Function]}
    onSave={[Function]}
    saving={false}
    user={Object {}}
  />,
]
`;
